### PR TITLE
Add option to create String() funcs

### PR DESCRIPTION
**Pull request by BenjaminTrapani**
Needed this for a https://clypdinc.atlassian.net/browse/MAIN-19181. Adds an optional flag to make the iota type implement stringer.

### CODE REVIEW

File: example/tostring_jsonenums.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		_toStringNameToValue = map[string]toString{
+			interface{}(elemA).(fmt.Stringer).String(): elemA,
+			interface{}(elemB).(fmt.Stringer).String(): elemB,
+			interface{}(elemC).(fmt.Stringer).String(): elemC,
+		}
+	}
+}
+
+func (r toString) String() string {
+	return _toStringValueToName[r]
```
Comment (jamesgarfield) [inline example/tostring_jsonenums.go]: Have we really thought through that this is how we want this to behave?

As it stands if someone makes an invalid value of the relevant type, we'll get an empty string out of this. In every other scenario we get an error.

Maybe that's ok, but I'm at least mildly concerned at the moment

Comment (BenjaminTrapani) [reply example/tostring_jsonenums.go]: @jamesgarfield We need the generated iota types to implement the Stringer interface, which requires a String() function with the given signature (no returned error). The generated code guarantees that all valid iota consts for the given type will be serialized successfully. This seems safe and transparent to me. What are your concerns with this approach?

Comment (quintenpalmer) [reply example/tostring_jsonenums.go]: One idea is to provide a method other than `String()` and let the consumer decide if they want the behavior they get from this to be what satisfies `Stringer`.

Comment (BenjaminTrapani) [reply example/tostring_jsonenums.go]: @quintenpalmer That sounds like a good plan. @jamesgarfield Does this sound good to you?

Comment (jamesgarfield) [reply example/tostring_jsonenums.go]: What would that look like?

In my head I'm seeing this

```
func (r toString) ToString() (string, error) {
   return r.getString()
}
```

And then if the consumer wants a `Stringer` they can call that and individually decide what to do with the error

Comment (quintenpalmer) [reply example/tostring_jsonenums.go]: That sounds reasonable to me @jamesgarfield.

**Comment by jamesgarfield** [comment]
👍

**Review by quintenpalmer — APPROVED** [review]
👍

**Comment by BenjaminTrapani** [comment]
@jamesgarfield @quintenpalmer The solution will not work, since there will be infinite recursion if we use the defined function in an external function which causes the type to implement Stringer (required for https://github.com/clyphub/gillnet/pull/4198). @jamesgarfield and I discussed this earlier. For simplicity I will use the current external implementation in https://github.com/clyphub/gillnet/pull/4198 unless @jamesgarfield no longer has objections to cd4330c.

**Comment by jamesgarfield** [comment]
@BenjaminTrapani couldn't we instead extract the common code from `getString` that doesn't include the path to default to `Stringer` and call that in `ToString`?

**Pull request merged**